lib/client/snapshot-request.js

```diff
@@ -0,0 +1,67 @@
+var hat = require('hat');
+var types = require('../types');
+
+module.exports = SnapshotRequest;
+
+function SnapshotRequest(connection, collection, id, callback) {
+  this.requestId = hat();
+
+  this.connection = connection;
+  this.id = id;
+  this.collection = collection;
+  this.callback = callback;
+
+  this.ready = false;
+  this.sent = false;
+}
+
+SnapshotRequest.prototype.hasPending = function () {
```

Comment:
`ready` and `hasPending` are not needed because all requests in `Connection.snapshotRequests` are pending.